test/Microsoft.AspNet.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -141,13 +141,11 @@ private class Person4
             Assert.NotNull(modelBindingResult);
             Assert.True(modelBindingResult.IsModelSet);
             var boundPerson = Assert.IsType<Person4>(modelBindingResult.Model);
-            Assert.NotNull(boundPerson);
-            Assert.False(modelState.IsValid);
 
-            // The error with an empty key is a bug(#2416)  in our implementation which does not append the prefix and
-            // use that along with the path. The expected key here would be CustomParameter.Address.
-            var key = Assert.Single(modelState.Keys, k => k == "");
-            var error = Assert.Single(modelState[""].Errors);
+            Assert.False(modelState.IsValid);
+            var entry = Assert.Single(modelState);
+            Assert.Equal(string.Empty, entry.Key);
+            var error = Assert.Single(entry.Value.Errors);
             Assert.StartsWith(
```

Comment:
Instead validate there was an error message (just check `!string.IsNullOrEmpty(Message)`) and add a comment about letting JSON.Net do whatever.